DokuwikiXMLExport.php

```diff
@@ -123,110 +168,78 @@ public function generateXMLExport($start, $count)
         return $exporter->serializeItems($items, $start, $total);
     }
 
-    protected $conf;
-
-    public function __construct($conf)
-    {
-        $this->conf = $conf;
-    }
-
-    /**
-     * Returns the ID / ordernumber.
-     *
-     * @param array $pages Contains all namespaces of all DokuWiki pages.
-     * @param integer $key The Item ID.
-     * @return string Returns the ordernumber.
-     */
-    private function getOrdernumber($pages, $key)
-    {
-        return $pages[$key];
-    }
-
     /**
      * Gets the Name of the current page.
      *
-     * @param array $pages Contains all namespaces of all DokuWiki pages.
-     * @param integer $key The Item ID.
      * @return string Returns the Name/Title of the page.
      */
-    private function getName($pages, $key)
+    private function getName()
     {
-        $metadata = p_get_metadata($pages[$key]);
+        $metadata = p_get_metadata($this->pages[$this->key]);
         return $metadata["title"];
     }
 
     /**
      * Gets the Summary and Description of the current page.
      *
-     * @param array $pages Contains all namespaces of all DokuWiki pages.
-     * @param integer $key The Item ID.
      * @return string Returns the Summary and Description of the page.
      */
-    private function getSummaryAndDescription($pages, $key)
+    private function getSummaryAndDescription()
     {
-        $metadata = p_get_metadata($pages[$key]);
+        $metadata = p_get_metadata($this->pages[$this->key]);
         return $metadata["description"]["abstract"];
     }
 
     /**
      * Gets the Url of the current page.
      *
-     * @param array $pages Contains all namespaces of all DokuWiki pages.
-     * @param integer $key The Item ID.
      * @return string Returns the Url of the page.
      */
-    private function getUrl($pages, $key)
+    private function getUrl()
     {
-        $url =  wl($pages[$key], '', true);
+        $url = wl($this->pages[$this->key], '', true);
         return $url;
     }
 
     /**
      * Gets the DateTime of the current page.
      *
-     * @param array $pages Contains all namespaces of all DokuWiki pages.
-     * @param integer $key The Item ID.
      * @return DateTime Returns the Date formatted in ATOM DateTime of the page.
      */
-    private function getDateAdded($pages, $key)
+    private function getDateAdded()
     {
-        $metadata = p_get_metadata($pages[$key]);
+        $metadata = p_get_metadata($this->pages[$this->key]);
         $date = new DateTime();
         $date->setTimestamp($metadata["date"]["created"]);
         return $date;
     }
 
     /**
+     * Returns the ID / ordernumber.
+     *
+     * @return string Returns the ordernumber.
+     */
+    private function getOrdernumber()
```

Comment:
I would name it `getId` or `getPageId` because it only becomes an ordernumber once it's put there in the export file.

Reply:
Totally agreed.